consensus/polybft/wallet/key.go

```diff
@@ -28,8 +29,14 @@ func (k *Key) Address() ethgo.Address {
 	return k.raw.Ecdsa.Address()
 }
 
-func (k *Key) Sign(b []byte) ([]byte, error) {
-	s, err := k.raw.Bls.Sign(b)
+// SignWithDomain signs message with bls and use bls.DomainCheckpointManager domain for mapping message to G1
+func (k *Key) Sign(hash []byte) ([]byte, error) {
```

Comment:
I would either:
- Rename this function to represent that it is signing with a `bls.DomainCheckpointManager`.
- Remove the function and make sure we do not miss use it.

Reply:
The function is needed because of `Key` interface
```
type Key interface {
	Address() Address
	Sign(hash []byte) ([]byte, error)
}
```

Reply:
Where is that function being used?

Reply:
I have realized that for Key implementation we only need signing with `bls.DomainCheckpointManager`.
So I have removed `func (k *Key) SignWithDomain`.  

  https://github.com/0xPolygon/polygon-edge/pull/1184/commits/de66a2c0e858b522f017b2550c91371d141c90f9
  
I

Reply:
Or maybe it would be even better if we have `Key` structure as
```
type Key struct {
	raw    *Account
	domain []byte
}

func NewKey(raw *Account, domain []byte) *Key {
	return &Key{
		raw:    raw,
		domain: domain,
	}
}

...

// Sign signs message with bls
func (k *Key) Sign(hash []byte) ([]byte, error) {
	s, err := k.raw.Bls.Sign(hash, k.domain)
	if err != nil {
		return nil, err
	}

	return s.Marshal()
}
```

Reply:
I did implementation like I suggested in comment above https://github.com/0xPolygon/polygon-edge/pull/1184/commits/a2639c7d6d84e00f3580fa04d0f7506cdf62dc83